src/hooks/useNote.js

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect } from 'react'
+
+function useNote({ serviceClient, noteId }) {
+  // `docId` is the id of the note for which we have a state.
+  // `noteId` is the id of the requested note.
+  const [docId, setDocId] = useState(noteId)
+  const [loading, setLoading] = useState(true)
+  const [doc, setDoc] = useState(undefined)
+  const [title, setTitle] = useState(undefined)
+
+  // reload if ever noteId changes
+  useEffect(
+    () => {
+      if (docId !== noteId) {
+        setLoading(true)
+        setTitle(undefined)
+        setDoc(undefined)
+        setDocId(noteId)
+      }
+    },
+    [noteId, docId, setLoading, setTitle, setDoc, setDocId]
+  )
+
+  // load the note
+  useEffect(
+    () => {
+      const loadNote = async function() {
+        try {
+          if (!loading) setLoading(true)
+          const doc = await serviceClient.getDoc(docId)
+          setTitle(doc.title || '')
+          setDoc(doc)
+        } catch (e) {
+          setTitle(false)
+          setDoc(false)
+          // eslint-disable-next-line no-console
+          console.warn(`Could not load note ${docId}`)
+        }
+        setLoading(false)
```

Comment:
Should this be in a `finally` block?

Reply:
The only difference I know of, is if the catch block is itself throwing an exception. If this happens here, I don't think we want to trigger an end of loading state.